gui/src/app/components/TabWidget.tsx

```diff
@@ -15,6 +15,17 @@ const TabWidget: FunctionComponent<TabWidgetProps> = ({ labels, children }) => {
   }
 
   const [index, setIndex] = useState(0);
+  const [tabsThatHaveBeenViewed, setTabsThatHaveBeenViewed] = useState<
+    number[]
```

Comment:
Thoughts on using a boolean[] of the same length as labels, rather than a number[]?